test/csv/interface/test_read.rb

```diff
@@ -32,6 +32,22 @@ def test_foreach
     assert_equal(@rows, rows)
   end
 
+  if RUBY_VERSION >= "2.7"
+    # Support to StringIO was dropped for Ruby 2.6 and earlier:
+    # https://github.com/ruby/stringio/pull/47
+    def test_foreach_stringio
+      string_io = StringIO.new(@data)
+      rows = CSV.foreach(string_io, col_sep: "\t", row_sep: "\r\n").to_a
+      assert_equal(@rows, rows)
+    end
+
+    def test_foreach_stringio_with_bom
+      string_io = StringIO.new("\ufeff#{@data}") # U+FEFF ZERO WIDTH NO-BREAK SPACE
+      rows = CSV.foreach(string_io, col_sep: "\t", row_sep: "\r\n").to_a
+      assert_equal(@rows, rows)
+    end
+  end
```

Comment:
Does `test_foreach_stringio` work with Ruby 2.7? If so, could you omit only `test_

```suggestion
  def test_foreach_stringio
    string_io = StringIO.new(@data)
    rows = CSV.foreach(string_io, col_sep: "\t", row_sep: "\r\n").to_a
    assert_equal(@rows, rows)
  end

  def test_foreach_stringio_with_bom
    if RUBY_VERSION < "2.7"
      # Support to StringIO was dropped for Ruby 2.6 and earlier without BOM support:
      # https://github.com/ruby/stringio/pull/47
      omit("StringIO's BOM support isn't available with Ruby < 2.7") 
    end
    string_io = StringIO.new("\ufeff#{@data}") # U+FEFF ZERO WIDTH NO-BREAK SPACE
    rows = CSV.foreach(string_io, col_sep: "\t", row_sep: "\r\n").to_a
    assert_equal(@rows, rows)
  end
```

Reply:
It does if I also initialize `StringIO` without additional options. I just pushed another commit like that. It's unfortunate having to run an additional `RUBY_VERSION` check at runtime, but at least that's only when calling `open`.